Source/JavaScriptCore/dfg/DFGOperations.cpp

```diff
@@ -2771,6 +2771,96 @@ JSC_DEFINE_JIT_OPERATION(operationStringLocaleCompare, UCPUStrictInt32, (JSGloba
     RELEASE_AND_RETURN(scope, toUCPUStrictInt32(collator->compareStrings(globalObject, string, that)));
 }
 
+JSC_DEFINE_JIT_OPERATION(operationStringIndexOf, UCPUStrictInt32, (JSGlobalObject* globalObject, JSString* base, JSString* argument))
+{
+    VM& vm = globalObject->vm();
+    CallFrame* callFrame = DECLARE_CALL_FRAME(vm);
+    JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
+
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto thisViewWithString = base->viewWithUnderlyingString(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    auto otherViewWithString = argument->viewWithUnderlyingString(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    size_t result = thisViewWithString.view.find(otherViewWithString.view);
+    if (result == notFound)
+        return toUCPUStrictInt32(-1);
+    return toUCPUStrictInt32(result);
+}
+
+JSC_DEFINE_JIT_OPERATION(operationStringIndexOfWithOneChar, UCPUStrictInt32, (JSGlobalObject* globalObject, JSString* base, int32_t character))
+{
+    VM& vm = globalObject->vm();
+    CallFrame* callFrame = DECLARE_CALL_FRAME(vm);
+    JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
+
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto thisViewWithString = base->viewWithUnderlyingString(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    size_t result = thisViewWithString.view.find(static_cast<UChar>(character));
+    if (result == notFound)
+        return toUCPUStrictInt32(-1);
+    return toUCPUStrictInt32(result);
+}
+
+JSC_DEFINE_JIT_OPERATION(operationStringIndexOfWithIndex, UCPUStrictInt32, (JSGlobalObject* globalObject, JSString* base, JSString* argument, int32_t position))
+{
+    VM& vm = globalObject->vm();
+    CallFrame* callFrame = DECLARE_CALL_FRAME(vm);
+    JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
+
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto thisViewWithString = base->viewWithUnderlyingString(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    auto otherViewWithString = argument->viewWithUnderlyingString(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    int32_t length = thisViewWithString.view.length();
+    unsigned pos = 0;
+    if (position >= 0)
```

Comment:
You only need `>` here.  Don't need `>=` because the minimum in that case must be `0`.

Reply:
I think `>=` is better since position < 0 is super rare, so >= is better for CPU prediction.